### PR TITLE
created front end for creating delivery employee

### PR DESCRIPTION
**Pull request by Adaray786**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import express from 'express';
 import nunjucks from 'nunjucks';
+import session from 'express-session';
 import path from 'path';
+import { DeliveryEmployee } from './model/deliveryEmployee';
+import exp from 'constants';
```
Comment (TomBillingtonUK) [inline src/app.ts]: is this import being used anywhere

File: src/model/deliveryEmployee.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export class DeliveryEmployee {
```
Comment (TomBillingtonUK) [inline src/model/deliveryEmployee.ts]: If this isn't being instantiated this could be a type

File: src/controller/deliveryEmployeeController.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { Application, Request, Response } from "express";
+import { DeliveryEmployee } from "../model/deliveryEmployee";
+
+const deliveryEmployeeService = require('../service/deliveryEmployeeService');
+
+module.exports = function(app: Application) {
```
Comment (TomBillingtonUK) [inline src/controller/deliveryEmployeeController.ts]: You could make this more typescript and user friendly by doing the following:

Replace module.exports with
```
export const deliveryEmployeeController = (app: Application) {
   ... code here
}
```

In app.ts you can import the function using:
`import {  deliveryEmployeeController } from 'path'`

You can then setup the routes by using
`deliveryEmployeeController(app)`

File: src/service/deliveryEmployeeService.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import axios from "axios";
+import e from "express";
```
Comment (TomBillingtonUK) [inline src/service/deliveryEmployeeService.ts]: do you need this import

File: src/service/deliveryEmployeeService.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { DeliveryEmployee } from "../model/deliveryEmployee";
+const deliveryEmployeeValidator = require('../validator/deliveryEmployeeValidator');
+
+module.exports.createDeliveryEmployee = async function(deliveryEmployee: DeliveryEmployee, token: string): Promise<number> {
```
Comment (TomBillingtonUK) [inline src/service/deliveryEmployeeService.ts]: Could you export this as per the example in the comment above

File: src/validator/deliveryEmployeeValidator.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { DeliveryEmployee } from "../model/deliveryEmployee";
+
+module.exports.validateDeliveryEmployee = function (deliveryEmployee: DeliveryEmployee):string | null {
```
Comment (TomBillingtonUK) [inline src/validator/deliveryEmployeeValidator.ts]: Could you export this as per the example above

File: src/views/add-deliveryEmployee-confirmation.html
Comment (TomBillingtonUK) [inline src/views/add-deliveryEmployee-confirmation.html:1]: Should this be an empty file?

File: src/views/add-deliveryEmployee-details.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
+<head>
```
Comment (TomBillingtonUK) [inline src/views/add-deliveryEmployee-details.html:1]: Screenshot of the page/pages please in the PR

File: src/controller/deliveryEmployeeController.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import { Application, Request, Response } from "express";
+import { DeliveryEmployee } from "../model/deliveryEmployee";
+
+const deliveryEmployeeService = require('../service/deliveryEmployeeService');
```
Comment (TomBillingtonUK) [inline src/controller/deliveryEmployeeController.ts]: Can you change this to an import

File: src/controller/deliveryEmployeeController.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
+import { Application, Request, Response } from "express";
+import { DeliveryEmployee } from "../model/deliveryEmployee";
```
Comment (TomBillingtonUK) [inline src/controller/deliveryEmployeeController.ts]: Can be imported as a type

File: src/service/deliveryEmployeeService.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+import { DeliveryEmployee } from "../model/deliveryEmployee";
+const deliveryEmployeeValidator = require('../validator/deliveryEmployeeValidator');
```
Comment (TomBillingtonUK) [inline src/service/deliveryEmployeeService.ts]: Please change this to a import instead of require

File: src/service/deliveryEmployeeService.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+import { DeliveryEmployee } from "../model/deliveryEmployee";
```
Comment (TomBillingtonUK) [inline src/service/deliveryEmployeeService.ts]: please import this as a type

File: src/service/deliveryEmployeeService.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+const deliveryEmployeeValidator = require('../validator/deliveryEmployeeValidator');
+
+export const createDeliveryEmployee = async function(deliveryEmployee: DeliveryEmployee, token: string): Promise<number> {
+    const error: string = deliveryEmployeeValidator.validateDeliveryEmployee(deliveryEmployee)
+
+    if (error) {
```
Comment (TomBillingtonUK) [inline src/service/deliveryEmployeeService.ts]: it would be better if the validator threw the error rather than us having an if statement here

**Review by TomBillingtonUK — COMMENTED** [review]
Still missing screenshots

**Review by TomBillingtonUK — APPROVED** [review]

**Review by RobertCrabtreeKainos — APPROVED** [review]
ig it lgtm

**Pull request merged**